nautobot/core/graphql/schema.py

```diff
@@ -378,22 +378,11 @@ def extend_schema_type_relationships(schema_type, model):
 
             # Generate the name of the attribute and the name of the resolver based on the slug of the relationship
             # and based on the prefix
-            rel_name = f"{prefix}{str_to_var_name(relationship.slug)}"
+            rel_name = f"{prefix}{str_to_var_name(relationship.key)}"
```

Comment:
If we assert correctness of the key as a clean/save requirement, we shouldn't need `str_to_var_name` here.